### PR TITLE
executor: fix unstable unit test TestIssue16696

### LINKED ISSUE #32040: Unstable test case testSerialSuite1.TestIssue16696
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
[2022-01-27T13:09:25.333Z] FAIL: sort_test.go:92: testSerialSuite1.TestIssue16696
[2022-01-27T13:09:25.333Z] 
[2022-01-27T13:09:25.333Z] sort_test.go:116:
[2022-01-27T13:09:25.333Z]     rows := tk.MustQuery("explain analyze  select t1.a, t1.a +1 from t t1 join t t2 join t t3 order by t1.a").Rows()
[2022-01-27T13:09:25.333Z] /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/util/testkit/testkit.go:419:
[2022-01-27T13:09:25.333Z]     tk.c.Check(err, check.IsNil, comment)
[2022-01-27T13:09:25.333Z] ... value *errors.withStack = open /tmp/1000_tidb/MC4wLjAuMDo0MDAwLzAuMC4wLjA6MTAwODA=/tmp-storage/chunk.ListInDisk-91113876942: no such file or directory ("open /tmp/1000_tidb/MC4wLjAuMDo0MDAwLzAuMC4wLjA6MTAwODA=/tmp-storage/chunk.ListInDisk-91113876942: no such file or directory")
[2022-01-27T13:09:25.333Z] ... sql:explain analyze  select t1.a, t1.a +1 from t t1 join t t2 join t t3 order by t1.a, args:[]
```

### 2. What did you expect to see? (Required)

No error

### 3. What did you see instead (Required)

Failed with error

This test seems to write a temporary file with fixed path, and some other test case also use this temporary file
When the test cases run parallelly, the temporary file is delete and case this failure.

### 4. What is your TiDB version? (Required)

07b0c7c657f4baf11f755e0ac7baf55bbc4c3bbd

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #32040

Problem Summary:

Some test case depends on the temporary file, while some other test case remove the temporary path directory.
When those tests run parallely, they are unstable.

### What is changed and how it works?

Use another temporary directory instead of the default one for some test cases.

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: executor/sort_test.go [hunk 1]
```diff
@@ -37,9 +37,12 @@ func TestSortInDisk(t *testing.T) {
 func testSortInDisk(t *testing.T, removeDir bool) {
 	restore := config.RestoreFunc()
 	defer restore()
+	tempPath, err := os.MkdirTemp(os.TempDir(), "*-tidb")
```
Comment (tisonkun) [inline executor/sort_test.go:40]: Maybe `t.TempDir()`

Comment (tiancaiamao) [reply executor/sort_test.go]: Done @tisonkun

**Review by tisonkun — APPROVED** [review]

**Pull request merged**